src/Internal/Payment/State/ProcessedState.php

```diff
@@ -65,7 +79,36 @@ public function complete_processing() {
 		$this->dpps->remove_session_processing_order( $order_id );
 		$this->order_service->update_order_from_successful_intent( $order_id, $context->get_intent(), $context );
 
+		// cleaning up.
+		wc_reduce_stock_levels( $order_id );
```

Comment:
This function should be called through the legacy service as well.

Reply:
Fixed